pkg/apis/core/types.go

```diff
@@ -5600,6 +5600,33 @@ const (
 	ScheduleAnyway UnsatisfiableConstraintAction = "ScheduleAnyway"
 )
 
+// NodeInclusionDirectivePolicy defines the type of node inclusion policy
+// +enum
+type NodeInclusionDirectivePolicy string
+
+const (
+	// Ignore means ignore this scheduling directive when calculating pod topology spread skew.
+	Ignore NodeInclusionDirectivePolicy = "Ignore"
+	// Respect means use this scheduling directive when calculating pod topology spread skew.
+	Respect NodeInclusionDirectivePolicy = "Respect"
```

Comment:
We are discussing replacing this word. `Ignore` is fine.

Alternatives for `Respect` could be:
- `Honor`
- `Filter`


Any thoughts @Huang-Wei ?

Reply:
I feel the term "Directive" doesn't make it more descriptive, probably just remove it.

BTW: can we follow the naming pattern of "TaintEffect" to name the type as "NodeInclusionEffect"? Also, a single term "Ignore" / "Respect" in `core` package are missing some contextual info, I'd prefer a bit of contextual naming like `NodeInclusionEffectIgnore` / `NodeInclusionEffectRespect`.

```go
NodeInclusionEffectIgnore NodeInclusionEffect = "Ignore"
NodeInclusionEffectRespect NodeInclusionEffect = "Respect"
```

> Alternatives for Respect could be:

`Honor` _slightly_ sounds better, but tbh I don't have a strong option...

Reply:
It can be `NodeInclusionPolicy` once the struct is removed.

I don't think Effect matches well.

Reply:
> I feel the term "Directive" doesn't make it more descriptive, probably just remove it.
> 
> BTW: can we follow the naming pattern of "TaintEffect" to name the type as "NodeInclusionEffect"? Also, a single term "Ignore" / "Respect" in `core` package are missing some contextual info, I'd prefer a bit of contextual naming like `NodeInclusionEffectIgnore` / `NodeInclusionEffectRespect`.
> 
> ```go
> NodeInclusionEffectIgnore NodeInclusionEffect = "Ignore"
> NodeInclusionEffectRespect NodeInclusionEffect = "Respect"
> ```
> 
> > Alternatives for Respect could be:
> 
> `Honor` _slightly_ sounds better, but tbh I don't have a strong option...

Good advices. Then I will also replace `Respect` with `Honor` then.